ir/function.cpp

```diff
@@ -289,4 +289,66 @@ void CFG::printDot(ostream &os) const {
   os << "}\n";
 }
 
+void DomTree::buildDominators() {
```

Comment:
what's the algorithm used here?

Reply:
The algorithm builds the dominators such that each dominator dominates itself and also contains the intersection of the dominators of all predecessors. It's just two formulas, it's better described here:
https://cs.stackexchange.com/questions/43105/dominator-tree-for-dag
This does require the CFG to be acyclic and that the edge iterator only visits one node after visiting all it's predecessors.